zebra-state/src/service/chain_tip.rs

```diff
@@ -320,6 +320,28 @@ impl ChainTipChange {
         Ok(action)
     }
 
+    /// Returns:
+    /// - `Some(`[`TipAction`]`)` if there has been a change since the last time the method was called.
+    /// - `None` if there has been no change.
+    ///
+    /// See [`wait_for_tip_change`] for details.
+    pub fn last_tip_change(&mut self) -> Option<TipAction> {
+        // Obtain the tip block.
+        let block_guard = self.receiver.borrow();
+        let block = block_guard.as_ref()?;
```

Comment:
Performance Nit: The borrow guard blocks the `watch::Sender` until the end of this method. That might slow down block commits in the state.

`last_tip_change` and `action` hold the lock longer than anywhere else. They doesn't do anything expensive yet, but that could change. 

So we're probably best to just clone immediately, and drop the lock. (`best_tip_block` does a clone, which is cheap. It's only ~90 bytes plus an `Arc` atomic increment.)

> Outstanding borrows hold a read lock. This means that long lived borrows could cause the send half to block. It is recommended to keep the borrow as short lived as possible.

https://docs.rs/tokio/0.3.7/tokio/sync/watch/struct.Receiver.html#method.borrow

```suggestion
        let block = self.best_tip_block()?;
```

Reply:
Please feel free to merge as soon as this is fixed - I don't need to review again.